eth/beacon/_utils/hash.py

```diff
@@ -9,3 +9,13 @@ def hash_eth2(data: bytes) -> Hash32:
     a future Ethereum 2.0 deployment phase.
     """
     return Hash32(keccak(data))
+
+
+def repeat_hash_eth2(data: bytes, repeats: int) -> Hash32:
+    """
+    Return Hash onion of ``data`` with ``repeats`` layers
+    """
+    if repeats == 0:
+        return Hash32(data)
+    else:
+        return repeat_hash_eth2(bytes(hash_eth2(data)), repeats - 1)
```

Comment:
Can you update this to the following implementation to get rid of the recursion.

```python
from eth_utils.toolz import pipe
import itertools

def repeat_hash_eth2(data, iterations):
    if iterations > 0:
        return pipe(data, *itertools.repeat(hash_eth2, iterations))
    elif iterations == 0:
        return data
    else:
        raise ValidationError("Iterations must be positive: Got {0}".format(iterations))
```

Also, is `iterations=0` a valid use case we expect to encounter?  If not, that code path should be removed.